File: ResourceManager/Models/Configurations/NoteTagConfiguration.vb
Imports Microsoft.EntityFrameworkCore.Metadata.Builders
Imports Microsoft.EntityFrameworkCore

Namespace ResourceManager.Models.Configurations
    Public Class NoteTagConfiguration
        Implements IEntityTypeConfiguration(Of NoteTag)
        Public Sub Configure(ByVal builder As EntityTypeBuilder(Of NoteTag)) Implements IEntityTypeConfiguration(Of NoteTag).Configure
            builder.ToTable("NoteTags")
            builder.HasKey(Function(x) New With {x.NoteID, x.TagID
            })

            builder.HasOne(Function(x) x.Note).WithMany(Function(x) x.NoteTags).HasForeignKey(Function(x) x.NoteID)

            builder.HasOne(Function(x) x.Tag).WithMany(Function(x) x.NoteTags).HasForeignKey(Function(x) x.TagID)

        End Sub
    End Class
End Namespace

File: ResourceManager/Models/Configurations/NoteTagConfiguration.cs
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using Microsoft.EntityFrameworkCore;
using System.Collections;
using System.Collections.Generic;
using ResourceManager.Configuration;

namespace ResourceManager.Models.Configurations
{
    public class NoteTagConfiguration : IEntityTypeConfiguration<NoteTag>
    {
        public void Configure(EntityTypeBuilder<NoteTag> builder)
        {
            builder.ToTable("NoteTags");
            builder.HasKey(x => new { x.NoteID, x.TagID });

            builder.HasOne(x => x.Note)
                .WithMany(x => x.NoteTags)
                .HasForeignKey(x => x.NoteID);

            builder.HasOne(x => x.Tag)
                .WithMany(x => x.NoteTags)
                .HasForeignKey(x => x.TagID);

        }
    }
}
